src/irmin-pack/s.ml

```diff
@@ -55,41 +55,72 @@ module type Specifics = sig
-      generation will be kept on disk after the gc finished. This option is
-      useful for debugging. The default is true.
+        If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
+        false immediately without launching a second gc. If [throttle] is
+        [Block] and there is a concurrent gc, [start_gc] blocks waiting for the
+        previous gc to finish and only after launches a second gc. If the
+        previous GC failed, the function returns without launching a new GC.
 
-      If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
-      false immediately without launching a second gc. If [throttle] is [Block]
-      and there is a concurrent gc, [start_gc] blocks waiting for the previous
-      gc to finish and only after launches a second gc. If the previous GC
-      failed, the function returns without launching a new GC.
+        TODO: Detail exceptions raised. *)
 
-      TODO: Detail exceptions raised. *)
+    val finalise_exn : ?wait:bool -> repo -> bool Lwt.t
+    (** [finalise_exn ?wait repo] waits for the gc process to finish in order to
+        finalise it. It returns true if a GC was finalised.
 
-  val finalise_gc : ?wait:bool -> repo -> bool Lwt.t
-  (** [finalise_gc ?wait repo] waits for the gc process to finish in order to
-      finalise it. It returns true if a GC was finalised.
+        Finalising consists of mutating [repo] so that it points to the new file
+        and to flush the internal caches that could be referencing GCed objects.
 
-      Finalising consists of mutating [repo] so that it points to the new file
-      and to flush the internal caches that could be referencing GCed objects.
+        If [wait = true] (the default), the call blocks until the GC process
+        finishes. If [wait = false] it either returns [false], raises an
+        exception or finalises and returns [true].
 
-      If [wait = true] (the default), the call blocks until the GC process
-      finishes. If [wait = false] it either returns [false], raises an exception
-      or finalises and returns [true].
+        If there are no running gcs, the call is a no-op and it returns false.
 
-      If there are no running gcs, the call is a no-op and it returns false.
+        TODO: Detail exceptions raised. *)
 
-      TODO: Detail exceptions raised. *)
+    (** {2 High-level API} *)
+
+    val run : repo -> commit_key -> bool Lwt.t
+    (** [run repo commit_key] attempts to start a GC process for a [repo] by
+        discarding all data prior to [commit_key]. Returns whether it started
+        successfully or not.
+
+        [run] will also finalise the GC process automaticlly. For more detailed
+        control, see {!start_exn} and {!finalise_exn}.
+
+        All exceptions that [Irmin_pack] knows how to handle are caught and
+        logged; others are re-raised. *)
+
+    val wait : repo -> bool Lwt.t
```

Comment:
What's the intended use for these 2 functions (wait / is_finished)? Are you thinking of monitor processes that are waiting for GC to finish to report the information and/or to schedule/block potential interfering IO operations?

I suggest changing the `bool` return here to something else (either `unit` or an abstract pretty-printable error).